packages/mermaid/src/diagrams/quadrant-chart/quadrantDb.ts

```diff
@@ -53,8 +54,61 @@ function setYAxisBottomText(textObj: LexTextObj) {
   quadrantBuilder.setData({ yAxisBottomText: textSanitizer(textObj.text) });
 }
 
-function addPoint(textObj: LexTextObj, x: number, y: number) {
-  quadrantBuilder.addPoints([{ x, y, text: textSanitizer(textObj.text) }]);
+function parseStyles(styles: string[]): StylesObject {
+  const stylesObject: StylesObject = {};
+  if (styles.length !== 0) {
+    for (const item of styles) {
+      const style = item.trim().split(/\s*:\s*/);
+      if (style[0] == 'radius') {
+        if (!/^\d+$/.test(style[1])) {
+          throw new Error(`value for radius ${style[1]} is unvalid, requires a number`);
+        }
+        stylesObject.radius = parseInt(style[1]);
+      } else if (style[0] == 'color') {
+        if (!/^#?([\dA-Fa-f]{6}|[\dA-Fa-f]{3})$/.test(style[1])) {
+          throw new Error(`value for color ${style[1]} is unvalid, requires a valid hex code`);
+        }
+        stylesObject.color = style[1];
+      } else if (style[0] == 'stroke-color') {
+        if (!/^#?([\dA-Fa-f]{6}|[\dA-Fa-f]{3})$/.test(style[1])) {
+          throw new Error(
+            `value for stroke-color ${style[1]} is unvalid, requires a valid hex code`
+          );
+        }
+        stylesObject.strokeColor = style[1];
+      } else if (style[0] == 'stroke-width') {
+        if (!/^\d+px$/.test(style[1])) {
+          throw new Error(
```

Comment:
```suggestion
          throw new InvalidStyleError(
```

And you can add a custom error class.